src/main/java/org/graylog/integrations/aws/service/KinesisService.java

```diff
@@ -243,10 +245,8 @@ private KinesisHealthCheckResponse handleCompressedMessages(KinesisHealthCheckRe
         Optional<CloudWatchLogEvent> logEntryOptional = data.logEvents().stream().findAny();
 
         if (!logEntryOptional.isPresent()) {
-            String message = "The CloudWatch payload did not contain any messages. This should not happen. " +
-                             "See https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/SubscriptionFilters.html";
-            LOG.debug(message);
-            return KinesisHealthCheckResponse.createFailed(message);
+            throw new BadRequestException("The CloudWatch payload did not contain any messages. This should not happen. " +
```

Comment:
It would be cleaner to refactor this code block to a method in CloudWatchService that returns a CloudWatchLogEvent object and just check for the type of payload in healthCheck, should it change in the future
```
         // Assume that the payload is from CloudWatch.
        // Extract messages, so that they can be committed to journal one by one.
        final CloudWatchLogSubscriptionData data = objectMapper.readValue(bytes, CloudWatchLogSubscriptionData.class);

        if (LOG.isTraceEnabled()) {
            // Log the number of events retrieved from CloudWatch. DO NOT log the content of the messages.
            LOG.trace("[{}] messages obtained from CloudWatch", data.logEvents().size());
        }

        // Pick just one log entry.
        Optional<CloudWatchLogEvent> logEntryOptional = data.logEvents().stream().findAny();

        if (!logEntryOptional.isPresent()) {
            throw new BadRequestException("The CloudWatch payload did not contain any messages. This should not happen. " +
                                          "See https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/SubscriptionFilters.html");
        }

        CloudWatchLogEvent logEntry = logEntryOptional.get();
```

Reply:
Great point. I 100% agree. I actually have a method prepared in #151 that can be reused for this.

Once #151 is reviewed/merged, I will use it here too. I created issue https://github.com/Graylog2/graylog-plugin-integrations/issues/173 to track this 

https://github.com/Graylog2/graylog-plugin-integrations/blob/cf1cf124d141348d265b803f3897ffa2803f4c2e/src/main/java/org/graylog/integrations/aws/transports/KinesisPayloadDecoder.java#L95-L110